acestep/core/generation/handler/prompt_utils.py

```
@@ -0,0 +1,162 @@
+            ]
+        )
+
+    def extract_caption_from_sft_format(self, caption: str) -> str:
+        """Extract caption body from SFT-formatted prompt when present."""
+        try:
+            if "# Instruction" in caption and "# Caption" in caption:
+                pattern = r"#\s*Caption\s*\n(.*?)(?:\n\s*#\s*Metas|$)"
+                match = re.search(pattern, caption, re.DOTALL)
+                if match:
+                    return match.group(1).strip()
+            return caption
```

Ruff (0.15.0)

[warning] Consider moving this statement to an `else` block

(TRY300)